yogstation/code/modules/antagonists/darkspawn/darkspawn_abilities/fighter_abilities.dm

```diff
@@ -0,0 +1,429 @@
+	check_flags = AB_CHECK_CONSCIOUS
+	panel = "Darkspawn"
+	antimagic_flags = NONE
+	spell_requirements = SPELL_REQUIRES_HUMAN
+	sound = 'yogstation/sound/creatures/darkspawn_howl.ogg'
+	psi_cost = 75
+
+/datum/action/cooldown/spell/time_dilation/can_cast_spell(feedback)
+	if(owner.has_status_effect(STATUS_EFFECT_TIME_DILATION))
+		if(feedback)
+			to_chat(owner, span_notice("You still have time dilation in effect."))
+		return FALSE
+	. = ..()
+
+/datum/action/cooldown/spell/time_dilation/cast(atom/cast_on)
+	. = ..()
+	var/mob/living/L = owner
+	L.apply_status_effect(STATUS_EFFECT_TIME_DILATION)
+	to_chat(L, span_velvet("Quix'thra ZYXAR!"))
+	L.visible_message(span_warning("[L] howls as their body sigils begin to scream light in every direction!"), span_velvet("Your sigils howl out light as your body moves at incredible speed!"))
+
+//////////////////////////////////////////////////////////////////////////
+//----------------------------Delayed AOE CC----------------------------//
+//////////////////////////////////////////////////////////////////////////
+/datum/action/cooldown/spell/aoe/demented_outburst
+	name = "Demented Outburst"
+	desc = "Deafens and confuses listeners after a five-second charge period, knocking away everyone nearby. Costs 50 Psi."
+	button_icon = 'yogstation/icons/mob/actions/actions_darkspawn.dmi'
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	buttontooltipstyle = "alien"
+	button_icon_state = "demented_outburst"
+	antimagic_flags = NONE
+	panel = "Darkspawn"
+	spell_requirements = SPELL_REQUIRES_HUMAN
+	check_flags = AB_CHECK_CONSCIOUS
+	psi_cost = 50 //big boom = big cost
+	aoe_radius = 5
+	var/casting = FALSE
+	var/cast_time = 5 SECONDS
+
+/datum/action/cooldown/spell/aoe/demented_outburst/can_cast_spell(feedback)
+	if(casting)
+		return FALSE
+	. = ..()
+
+/datum/action/cooldown/spell/aoe/demented_outburst/before_cast(atom/cast_on)
+	. = ..()
+	if(casting)
+		return . | SPELL_CANCEL_CAST
+	if(. & SPELL_CANCEL_CAST)
+		return .
+	casting = TRUE
+	to_chat(owner, span_velvet("Kap..."))
+	owner.visible_message(span_boldwarning("[owner] begins to growl as their chitin hardens..."), span_velvet("You begin focusing your power..."))
+	playsound(owner, 'yogstation/sound/magic/demented_outburst_charge.ogg', 50, 0)
+	if(!do_after(owner, cast_time, cast_on))
+		casting = FALSE
+		return . | SPELL_CANCEL_CAST
+	casting = FALSE
+
+/datum/action/cooldown/spell/aoe/demented_outburst/cast(atom/cast_on)
+	. = ..()
+	to_chat(owner, span_velvet("...WXSU!"))
+	owner.visible_message(span_userdanger("[owner] lets out a deafening scream!"), span_velvet("You let out a deafening outburst!"))
+	playsound(owner, 'yogstation/sound/magic/demented_outburst_scream.ogg', 75, 0)
+
+/datum/action/cooldown/spell/aoe/demented_outburst/cast_on_thing_in_aoe(atom/victim, atom/caster)
+	if(!can_see(victim, caster, aoe_radius))
+		return
+	if(!ismovable(victim))
+		return
+	var/atom/movable/AM = victim
```

Comment:
please use better var name than ``AM``. You can just path victim to movable and use that, you already check right above this for ismovable.